src/components/Menu/MenuBar.vue

```diff
@@ -161,12 +162,19 @@ export default {
 			return list
 		},
 		hiddenEntries() {
-			const entries = this.entries.filter(({ priority }) => {
+			const remainingEntries = this.entries.filter(({ priority }) => {
 				// reverse logic from visibleEntries
 				return priority !== undefined && priority > this.iconsLimit
-			}).reduce((acc, entry) => {
+			})
+			const entries = remainingEntries.reduce((acc, entry, index) => {
 				// If entry has children, merge them into list. Otherwise keep entry itself.
 				const children = entry.children ?? [entry]
+				// If this block has menu entries, it should be separated for better visibility and a11y (menu item radio grouping)
+				if (children.length > 1) {
+					const separatorBefore = acc.length ? [{ key: `separator-before-${entry.id}`, isSeparator: true }] : []
+					const separatorAfter = index !== remainingEntries.length - 1 ? [{ key: `separator-after-${entry.id}`, isSeparator: true }] : []
+					return [...acc, ...separatorBefore, ...children, ...separatorAfter]
+				}
```

Comment:
If i parsed this correctly it may result in duplicate separators as soon as two entries with children are next to each other. I don't know if this can happen currently nor if it would be a problem. Will investigate some more.

Reply:
Nice catch

Reply:
Added a fixup

Reply:
Thanks! Looks good. I was not able to reproduce this as we never move the headings into the 'more' menu - but that might change.